lib/helpers.js

```diff
@@ -33,17 +32,21 @@ function buildOnSettled(done) {
       return done(error, null);
     }
 
-    var settledErrors = filter(result, filterError);
-    var settledResults = filter(result, filterSuccess);
+    if (result == null) {
```

Comment:
```suggestion
    if (!Array.isArray(result)) {
```
I think we want to always check that it is an array, right?